src/components/textbox/textbox.test.js

```diff
@@ -0,0 +1,712 @@
+      CypressMountWithProviders(<TextboxComponent onClick={callback} />);
+
+      textboxInput()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onFocus callback when a focus event is triggered", () => {
+      CypressMountWithProviders(<TextboxComponent onFocus={callback} />);
+
+      textboxInput()
+        .focus()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onMouseDown callback when a mousedown event is triggered", () => {
+      CypressMountWithProviders(<TextboxComponent onMouseDown={callback} />);
+
+      textboxInput()
+        .trigger("mousedown")
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call iconOnMouseDown callback when a click event is triggered", () => {
+      CypressMountWithProviders(
+        <TextboxComponent iconOnMouseDown={callback} inputIcon="add" />
+      );
+
+      getComponent("icon")
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call iconOnClick callback when a click event is triggered", () => {
+      CypressMountWithProviders(
+        <TextboxComponent iconOnClick={callback} inputIcon="add" />
+      );
+
+      getComponent("icon")
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+    it.each([["Enter"], ["Space"]])(
+      "should call iconOnClick callback when %s key is triggered",
+      (key) => {
+        CypressMountWithProviders(
+          <TextboxComponent
+            inputIcon="home"
+            iconOnClick={callback}
+            iconTabIndex="0"
+          />
+        );
+
+        getComponent("icon")
+          .trigger("keydown", { ...keyCode(key), force: true })
+          .then(() => {
+            // eslint-disable-next-line no-unused-expressions
+            expect(callback).to.have.been.calledOnce;
+          });
+      }
+    );
+
+    it.each([["Enter"], ["Space"]])(
+      "should call onKeyDown callback when %s key is triggered",
+      (key) => {
+        CypressMountWithProviders(<TextboxComponent onKeyDown={callback} />);
+
+        textboxInput()
+          .focus()
+          .trigger("keydown", { ...keyCode(key), force: true })
+          .then(() => {
+            // eslint-disable-next-line no-unused-expressions
+            expect(callback).to.have.been.calledOnce;
+          });
+      }
+    );
+
+    it.each([
+      [1000, "1"],
+      [5000, "5"],
+      [10000, "10"],
```

Comment:
In this test case, do we need to include a bit longer time, for example [50000, "50"]?

Reply:
why? how 10s is different from 50s?